### PR TITLE
Test improvements

### PR DESCRIPTION
**Pull request by cjellick**
Introduce the kubernetes python client to allow tests to query
kubernetes directly

Add some more helper functions to conftest

Introduce a project-owner rbac test

### CODE REVIEW

**Comment by nathan-jenan-rancher** [comment]
LGTM

File: tests/core/conftest.py [hunk 1]
```diff
@@ -58,19 +64,35 @@ def admin_mc():
     }, verify=False)
     client = rancher.Client(url=BASE_URL, token=r.json()['token'],
                             verify=False)
-    return ManagementContext(client)
+    k8s_client = kubernetes_api_client(client, 'local')
+    return ManagementContext(client, k8s_client)
+
+
+def kubernetes_api_client(rancher_client, cluster_name):
```
Comment (dramich) [inline tests/core/conftest.py]: Can we keep the fixtures at the top and functions after that? Class > fixture > func

File: tests/core/conftest.py [hunk 2]
```diff
@@ -107,7 +129,30 @@ def user_mc(admin_mc):
     }, verify=False)
     client = rancher.Client(url=BASE_URL, token=response.json()['token'],
                             verify=False)
-    return ManagementContext(client)
+    return ManagementContext(client, user=user)
+
+
+def wait_until_available(client, obj):
+    timeout = 45
```
Comment (dramich) [inline tests/core/conftest.py]: `timeout` should be passed in with a default of 45.

File: tests/core/conftest.py [hunk 3]
```diff
@@ -58,19 +64,35 @@ def admin_mc():
...
+    k8s_client = ApiClient(configuration=client_configuration)
+    return k8s_client
 
 
 @pytest.fixture
 def admin_cc(admin_mc):
     """Returns a ClusterContext for the local cluster for the default global
     admin user."""
-    cluster = admin_mc.client.by_id_cluster('local')
+    cluster, client = cluster_and_client('local', admin_mc.client)
```
Comment (dramich) [inline tests/core/conftest.py]: `local` should end up as a const, we use it too much then we can set it later with an env var

File: tests/core/conftest.py (hunk 1 above)
Comment (nathan-jenan-rancher) [reply tests/core/conftest.py]: Are there linters that will verify this for us?

**Comment by cjellick** [comment]
Addresses
https://github.com/rancher/rancher/issues/13703
https://github.com/rancher/rancher/issues/14263 (I may add more tests in the future, but this gets the ball rolling and its all I'll do for now)

File: tests/core/conftest.py (hunk 3 above)
Comment (cjellick) [reply tests/core/conftest.py]: lets address this one when we integrate with hussein's work to pull in secondary cluster

**Comment by dramich** [comment]
lgtm

**Pull request merged**